File: rebate/src/main/java/com/binance4j/rebate/dto/RebateData.java
package com.binance4j.rebate.dto;

import java.util.List;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

/**
 * Rebate data.
 * 
 * @property page         Current page.
 * @property totalRecords Total records.
 * @property totalPageNum Total pages.
 * @property data         Rebates.
 */
@ApiModel("Rebate data.")
data class RebateData(
@ApiModelProperty("Current page.")
var page:int?=null,
@ApiModelProperty("Total records.")
var totalRecords:int?=null,
@ApiModelProperty("Total pages.")
var totalPageNum:int?=null,
@ApiModelProperty("Rebates.") List<Rebate> data)
{
}

File: connectors/src/main/kotlin/com/binance4j/rebate/dto/RebateData.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.rebate.dto

import com.fasterxml.jackson.annotation.JsonProperty

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * Rebate data.
 *
 * @property page         Current page.
 * @property totalRecords Total records.
 * @property totalPageNum Total pages.
 * @property data         Rebates.
 */
@ApiModel("Rebate data.")
data class RebateData(
	@ApiModelProperty("Current page.")
	@JsonProperty("page")
	var page: Int,
	@ApiModelProperty("Total records.")
	@JsonProperty("totalRecords")
	var totalRecords: Int,
	@ApiModelProperty("Total pages.")
	@JsonProperty("totalPageNum")
	var totalPageNum: Int,
	@ApiModelProperty("Rebates.")
	@JsonProperty("data")
	var data: List<Rebate>
)